lesson_22/listManipulation.js

```diff
@@ -0,0 +1,34 @@
+import { clearSelected, addSelected } from "./helpers.js";
+
+function listManipulation(e) {
+  if (e.target === this) {
+    return false;
+  }
+  if (e.ctrlKey) {
+    e.target.classList.toggle("selected");
+  } else if (e.shiftKey) {
+    let selected = this.querySelector(".selected");
+    if (selected) {
+      let start = Array.from(this.children).indexOf(selected);
+      let end = Array.from(this.children).indexOf(e.target);
+      let min = Math.min(start, end);
+      let max = Math.max(start, end);
+      for (let i = min; i <= max; i++) {
+        this.children[i].classList.add("selected");
+      }
+    } else {
+      let start = Array.from(this.children).indexOf(this.firstElementChild);
+      let end = Array.from(this.children).indexOf(e.target);
+      let min = Math.min(start, end);
+      let max = Math.max(start, end);
+      for (let i = min; i <= max; i++) {
+        this.children[i].classList.add("selected");
+      }
+    }
```

Comment:
1. shiftKey condition need to be a separate function selectRange(). Think about what arguments it should have.
2. This code can be refactored and it can become 2 times shorter. Think about how to do it